core/trino-main/src/main/java/io/trino/execution/CreateSchemaTask.java

```diff
@@ -114,11 +114,12 @@ public String explain(CreateSchema statement, List<Expression> parameters)
         return immediateVoidFuture();
     }
 
-    private TrinoPrincipal getCreatePrincipal(CreateSchema statement, Session session, Metadata metadata)
+    private TrinoPrincipal getCreatePrincipal(CreateSchema statement, CatalogSchemaName schema, Session session, Metadata metadata)
```

Comment:
wouldn't it be better to pass catalogName here, instead of passing the schema-yet-to-be-created?